src/Microsoft.DncEng.SecretManager/Readme.md

```diff
@@ -0,0 +1,196 @@
+
+## Creating new secrets
+New secrets can be created by adding new entries to the manifest file, then running synchronize. The tool will then create the new secret and add it to the secret store.
+
+## Manifest Files
+A manifest file describes the expected state of a single secret store (currently only azure key vault, but support for other stores can be added easily). The manifest contains an inventory of all secrets and the information required to generate/rotate them. The format is as follows.
+```yaml
+# The destination secret store. All the secrets described in this document will be placed in here.
+storageLocation:
+  type: azure-key-vault
+  parameters:
+    name: <a string>
+    subscription: <a guid>
+
+# A collection of named references to other secret stores, secrets in this document can reference values from these stores if they are required
+references:
+  reference1:
+    type: azure-key-vault
+    parameters:
+      name: <a string>
+      subscription: <a guid>
+
+
+# encryption keys that are needed in the vault, these keys will be created if they don't exist.
+keys:
+  encryption-key-one:
+    type: RSA
+    size: 2048
+
+# each entry below is a separate secret that will be maintained in the secret store
+# the type identifies what kind of secret it is, and the parameters drive the creation of the secret
+secrets:
+  secret1:
+    type: pizza-recipe
+    parameters:
+      # This is a 'SecretReference' parameter. Some secrets require other secrets to work.
+      # These parameters can be specified as a raw string, or with explicit name and location
+      # If a raw string is used, the referenced secret is assumed to be in the same vault as the current secret
+      # location can be any of the entries in the 'references' collection
+      doughToken:
+        name: dough-provider-token
+        location: reference1
+      # other parameters are just strings, and their interpretation depends on the type of the secret
+      cheese: cheddar
+      sauce: marinara
+      topping: pepperoni
+```
+
+## Secret Types
+Each secret in the manifest has a specified type, and these types determine what parameters are required, and what ends up in the secret store. Some secret types require human interaction to generate and/or rotate. Some secret types also produce multiple values. These values are all stored with additional suffixes in the secret store. Each type is documented below.
+
+### Azure Storage
+
+#### Connection String
+```yaml
+type: azure-storage-connection-string
+parameters:
+  subscription: azure subscription id
```

Comment:
Do these subscription IDs need to be the GUID, or is the friendly name acceptable?

Reply:
They must be the guid.